### DISCUSSION TITLE
RollingFileAppender and how rotation is handled

### DISCUSSION THREAD

**Discussion by Zerrossetto** [qos-ch/logback#656] [General]
Hi,
we're currently using logback 1.3.5 on several applications in our company, and I have a technical question regarding file rolling.

Is there someplace that describes how RollingFileAppender handles log files rolling? On an high-level perspective it should be enough understanding the order in which file is renamed, stream to the old active file is closed and then re-opened.
A pointer wher to read on this topic would be very much welcomed.

**Comment by ceki** [comment]
Hello @Zerrossetto, 

Please see [ch.qos.logback.core.rolling.RollingFileAppender#rollover](https://github.com/qos-ch/logback/blob/ec508c48b2e33662be69349ac9b784ab7d292df1/logback-core/src/main/java/ch/qos/logback/core/rolling/RollingFileAppender.java#LL189C17-L189C25)

If compression is needed, it happens asynchronously for time based compression. See ch.qos.logback.core.rolling.TimeBasedRollingPolicy#rollover

For FixedWindowRollingPolicy, compression, if requested, happens synchronously.
